### PR TITLE
Adding writing and coding snippet about location listening weakreference setup

### LINKED ISSUE #834: Add mention about LocationEngine leak prevention
Ticket to track @Guardiola31337 's suggestion that we add a comment in  https://docs.mapbox.com/android/core/overview/#locationengine on how to set up the location engine  to prevent leaks.  

cc @electrostat @andrlee 

- https://github.com/mapbox/mapbox-events-android/issues/289 
- https://github.com/mapbox/mapbox-events-android/pull/290#issuecomment-441248125 
- https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java#L1245-L1264 
- https://github.com/mapbox/mapbox-navigation-android/pull/1615/files#diff-f70627b24d5b0066cef281743eae0f6eR66

### PR DESCRIPTION
**Pull request by langsmith**
Resolves #834 by adding writing and code snippets related to `WeakReference` location listening callback.

screencapture-localhost-8080-android-core-overview-2019-02-08-15_00_06

### CODE REVIEW

**Comment by langsmith** [comment]
cc @Guardiola31337 , who originally suggested this addition to our docs site.

File: src/pages/core/overview/index.md [hunk 1]
```diff
@@ -192,152 +192,153 @@ var locationEngine = LocationEngineProvider.getBestLocationEngine(this)
 
 This will obtain the best location engine that is available and eliminate the need to create a new `LocationEngine` from scratch.
 
-### Getting location updates
+## Requesting location updates
+
+You'll need a class which implements `LocationEngineCallback<LocationEngineResult>`. Make sure the class requires Android system `Activity` as a constructor parameter. This class will serve as a "callback" and it's needed because a `LocationEngine` memory leak is possible if the activity/fragment directly implements the `LocationEngineCallback<LocationEngineResult>`. The `WeakReference` setup avoids the leak.
```
Comment (colleenmcginnis) [inline src/pages/core/overview/index.md]: `which` --> `that` 😉

File: src/pages/core/overview/index.md [hunk 2]
```diff
@@ -192,152 +192,153 @@ var locationEngine = LocationEngineProvider.getBestLocationEngine(this)
...
 
-Add the `mapbox-android-core` dependency to your project in order to listen to location changes. Then initialize a new instance of `LocationEngine` as described above. Once it's created, activate the `LocationEngine` object. Either the parent Activity should implement `LocationEngineCallback` or a separate `LocationEngineCallback` object to include in your update request. Implementing the `LocationEngineCallback` interface will require you to override the `onSuccess()` and `onFailure()` methods. Inside of `onSuccess()`, you will receive a `LocationEngineResult` which contains location results from the engine.
+Implementing the `LocationEngineCallback` interface will require you to override the `onSuccess()` and `onFailure()` methods.`OnSuccess()` runs whenever the Mapbox Core Libraries identifies a change in the device's location. `result.getLastLocation()` gives you a `Location` object and that object has the latitude and longitude values. Now you can display the values in your app's UI, save it in memory, send it to your backend server, or use the device location information however you'd like.
```
Comment (colleenmcginnis) [inline src/pages/core/overview/index.md]: >Implementing the `LocationEngineCallback` interface will require you to override the `onSuccess()` and `onFailure()` methods.

maybe reword

>When implementing the `LocationEngineCallback` interface, you are also required to override the `onSuccess()` and `onFailure()` methods.

Comment (colleenmcginnis) [inline src/pages/core/overview/index.md]: I think you're missing a space after this period: methods.`OnSuccess()\`

Comment (colleenmcginnis) [inline src/pages/core/overview/index.md]: Simplify 

>`result.getLastLocation()` gives you a `Location` object and that object has the latitude and longitude values.

to 

>`result.getLastLocation()` gives you a `Location` object that contains the latitude and longitude values.

File: src/pages/core/overview/index.md (hunk 1 above)
Comment (langsmith) [reply src/pages/core/overview/index.md]: fixed. I think I finally get it.

File: src/pages/core/overview/index.md (hunk 2 above)
Comment (langsmith) [reply src/pages/core/overview/index.md]: Done

Comment (langsmith) [reply src/pages/core/overview/index.md]: Added

Comment (langsmith) [reply src/pages/core/overview/index.md]: Fixed

**Comment by langsmith** [comment]
Don't merge until @electrostat gets a chance to review
